File: app/src/main/java/worker/AdacaWorker.java
package worker;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import java.util.Objects;

import model.Game;
import webservice.MemoriaRepository;

public class AdacaWorker extends Worker {
    private final MemoriaRepository repository = MemoriaRepository.getInstance();

    public AdacaWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);
    }

    @NonNull
    @Override
    public Result doWork() {
        try {
            Game game = Game.parseGame(Objects.requireNonNull(getInputData().getString("game")));
            repository.gravarDados(game.getDeviceUID(), game, getApplicationContext());

            return Result.success();
        } catch (NullPointerException exception) {
            return Result.failure();
        } catch (Exception exception) {
            return Result.retry();
        }
    }
}

File: app/src/main/kotlin/worker/AdacaWorker.kt
package worker

import android.content.Context
import androidx.work.Worker
import androidx.work.WorkerParameters
import model.Game
import webservice.MemoriaRepository

class AdacaWorker(context: Context, workerParams: WorkerParameters) : Worker(context, workerParams) {
    private var repository: MemoriaRepository = MemoriaRepository.getInstance()

    override fun doWork(): Result {
        return try {
            val gameInputData = inputData.getString("game")
            if (gameInputData != null) {
                val game = Game.parseGame(gameInputData)
                repository.writeData(game.deviceUID, game, applicationContext)

                Result.success()
            } else throw NullPointerException()
        } catch (exception: NullPointerException) {
            return Result.failure()
        } catch (exception: Exception) {
            Result.retry()
        }
    }
}